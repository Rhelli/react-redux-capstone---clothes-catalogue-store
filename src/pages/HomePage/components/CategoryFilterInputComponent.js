import React from 'react';
import PropTypes from 'prop-types';
import CategoryTagsComponent from './CategoryTagsComponent';
import FilterTagsComponent from './FilterTagsComponent';

const CategoryFilterInputComponent = props => {
  const {
    categoryData, enableSelectedFilter,
  } = props;

  return (
    <div>
      <div>
        <CategoryTagsComponent
          enableSelectedFilter={enableSelectedFilter}
          categoryData={categoryData}
        />
      </div>
      <div>
        <FilterTagsComponent
          enableSelectedFilter={enableSelectedFilter}
        />
      </div>
    </div>
  );
};

CategoryFilterInputComponent.propTypes = {
  enableSelectedFilter: PropTypes.func.isRequired,
  categoryData: PropTypes.shape({
    clothesFilter: PropTypes.shape({
      jackets: PropTypes.bool.isRequired,
      jeans: PropTypes.bool.isRequired,
      shirts: PropTypes.bool.isRequired,
      tops: PropTypes.bool.isRequired,
      dresses: PropTypes.bool.isRequired,
    }).isRequired,
    genderFilter: PropTypes.shape({
      male: PropTypes.bool.isRequired,
      female: PropTypes.bool.isRequired,
    }).isRequired,
    colorFilter: PropTypes.shape({
      black: PropTypes.bool.isRequired,
      darkestSpruce: PropTypes.bool.isRequired,
      multi: PropTypes.bool.isRequired,
      blue: PropTypes.bool.isRequired,
      blueDenim: PropTypes.bool.isRequired,
      navy: PropTypes.bool.isRequired,
      green: PropTypes.bool.isRequired,
      white: PropTypes.bool.isRequired,
      grape: PropTypes.bool.isRequired,
      khaki: PropTypes.bool.isRequired,
      grey: PropTypes.bool.isRequired,
      vintageIndigo: PropTypes.bool.isRequired,
      mineralBlue: PropTypes.bool.isRequired,
      charcoal: PropTypes.bool.isRequired,
      tan: PropTypes.bool.isRequired,
      dustyBlue: PropTypes.bool.isRequired,
      midwash: PropTypes.bool.isRequired,
      red: PropTypes.bool.isRequired,
      rose: PropTypes.bool.isRequired,
      lilac: PropTypes.bool.isRequired,
    }).isRequired,
  }).isRequired,
};

export default CategoryFilterInputComponent;
