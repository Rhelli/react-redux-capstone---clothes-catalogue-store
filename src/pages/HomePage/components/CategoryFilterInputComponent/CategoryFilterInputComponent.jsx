import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilter } from '@fortawesome/free-solid-svg-icons';
import CategoryTagsComponent from '../CategoryTagsComponent/CategoryTagsComponent';
import FilterTagsComponent from '../FilterTagsComponent/FilterTagsComponent';
import styles from './CategoryFilterInputComponent.module.scss';

const CategoryFilterInputComponent = props => {
  const {
    categoryData, enableSelectedFilter,
  } = props;

  return (
    <div className={styles.filterInputContainer}>
      <h3>
        Filters&nbsp;
        <FontAwesomeIcon className={styles.filterIcon} icon={faFilter} />
      </h3>
      <div className={styles.filterInputDropdown}>
        <div>
          <FilterTagsComponent
            enableSelectedFilter={enableSelectedFilter}
          />
        </div>
        <div>
          <CategoryTagsComponent
            enableSelectedFilter={enableSelectedFilter}
            categoryData={categoryData}
          />
        </div>
      </div>
    </div>
  );
};

CategoryFilterInputComponent.propTypes = {
  enableSelectedFilter: PropTypes.func.isRequired,
  categoryData: PropTypes.shape({
    clothesFilter: PropTypes.shape({
      jacket: PropTypes.bool.isRequired,
      jeans: PropTypes.bool.isRequired,
      shirts: PropTypes.bool.isRequired,
      tops: PropTypes.bool.isRequired,
      dresses: PropTypes.bool.isRequired,
    }).isRequired,
    genderFilter: PropTypes.shape({
      male: PropTypes.bool.isRequired,
      female: PropTypes.bool.isRequired,
    }).isRequired,
    colorFilter: PropTypes.shape({
      black: PropTypes.bool.isRequired,
      darkestSpruce: PropTypes.bool.isRequired,
      multi: PropTypes.bool.isRequired,
      blue: PropTypes.bool.isRequired,
      blueDenim: PropTypes.bool.isRequired,
      navy: PropTypes.bool.isRequired,
      green: PropTypes.bool.isRequired,
      white: PropTypes.bool.isRequired,
      grape: PropTypes.bool.isRequired,
      khaki: PropTypes.bool.isRequired,
      grey: PropTypes.bool.isRequired,
      vintageIndigo: PropTypes.bool.isRequired,
      mineralBlue: PropTypes.bool.isRequired,
      charcoal: PropTypes.bool.isRequired,
      tan: PropTypes.bool.isRequired,
      dustyBlue: PropTypes.bool.isRequired,
      midwash: PropTypes.bool.isRequired,
      red: PropTypes.bool.isRequired,
      rose: PropTypes.bool.isRequired,
      lilac: PropTypes.bool.isRequired,
    }).isRequired,
  }).isRequired,
};

export default CategoryFilterInputComponent;
