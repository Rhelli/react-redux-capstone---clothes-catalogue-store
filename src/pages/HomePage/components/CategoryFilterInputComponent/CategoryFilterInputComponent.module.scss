.filterInputContainer {
  margin: 10px 0 10px 0;
  position: relative;
  display: inline-block;

  &:hover .filterInputDropdown {
    display: block;
  }

  h3 {
    padding: 6px 16px;
    font: 400 20px/20px 'Source Sans Pro', monospace;
  }
}

.filterIcon {
  font-size: 14px;
  color: #303030;
  margin: 0 0 1px 0;
}

.filterInputDropdown {
  display: none;
  position: absolute;
  margin-top: 5px;
  width: 100vw;
  z-index: 1;
  background-color: #fdf2e9;
  border-top: 1px solid #ede4dc;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}
