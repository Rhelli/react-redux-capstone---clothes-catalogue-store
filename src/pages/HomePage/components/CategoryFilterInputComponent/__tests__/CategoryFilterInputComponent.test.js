import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import CategoryFilterInputComponent from '../CategoryFilterInputComponent';

const enableSelectedFilter = jest.fn();
const categoryData = {
  clothesFilter: {
    jacket: true,
    jeans: false,
    shirts: true,
    tops: false,
    dresses: true,
  },
  genderFilter: {
    male: false,
    female: true,
  },
  colorFilter: {
    black: false,
    darkestSpruce: true,
    multi: false,
    blue: true,
    blueDenim: false,
    navy: true,
    green: false,
    white: true,
    grape: false,
    khaki: true,
    grey: false,
    vintageIndigo: true,
    mineralBlue: false,
    charcoal: true,
    tan: false,
    dustyBlue: true,
    midwash: false,
    red: true,
    rose: false,
    lilac: true,
  },
};

const categoryFilterInputComponent = (
  <CategoryFilterInputComponent
    categoryData={categoryData}
    enableSelectedFilter={enableSelectedFilter}
  />
);

describe('The appearance and functionality of the CategoryFilterInputComponent', () => {
  it('Matches the CategoryFilterInputComponents snapshot', () => {
    const tree = renderer.create(categoryFilterInputComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains a master container', () => {
    const { container } = render(categoryFilterInputComponent);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('filterInputContainer')).toBe(true);
  });

  it('Contains the filter title and the FontAwesome Icon', () => {
    const { container, getByText } = render(categoryFilterInputComponent);
    expect(container.firstChild.firstChild.nodeName).toBe('H3');
    getByText('Filters');
  });

  it('Contains a hidden dropdown menu upon startup', () => {
    const { container } = render(categoryFilterInputComponent);
    expect(container.firstChild.lastChild.visibility).toBeFalsy();
  });
});
