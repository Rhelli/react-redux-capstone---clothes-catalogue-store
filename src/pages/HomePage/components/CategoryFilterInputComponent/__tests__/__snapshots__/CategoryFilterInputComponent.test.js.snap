// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the CategoryFilterInputComponent Matches the CategoryFilterInputComponents snapshot 1`] = `
<div
  className="filterInputContainer"
>
  <h3>
    FiltersÂ 
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-filter fa-w-16 filterIcon"
      data-icon="filter"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </h3>
  <div
    className="filterInputDropdown"
  >
    <div>
      <div
        className="filterTagsContainer"
      >
        <div
          className="filterCategory"
        >
          <h4
            className="categoryTitles"
          >
            Gender
          </h4>
          <div
            className="categoryTags"
          >
            <div
              className="male"
              data-name="male"
              onClick={[Function]}
            >
              <h6
                data-name="male"
              >
                Men's
              </h6>
            </div>
            <div
              className="female"
              data-name="female"
              onClick={[Function]}
            >
              <h6
                data-name="female"
              >
                Women's
              </h6>
            </div>
          </div>
        </div>
        <div
          className="filterCategory"
        >
          <h4
            className="categoryTitles"
          >
            Clothes
          </h4>
          <div
            className="categoryTags"
          >
            <div
              className="jacket"
              data-name="jacket"
              onClick={[Function]}
            >
              <h6
                data-name="jacket"
              >
                Coats & Jackets
              </h6>
            </div>
            <div
              className="jeans"
              data-name="jeans"
              onClick={[Function]}
            >
              <h6
                data-name="jeans"
              >
                Jeans
              </h6>
            </div>
            <div
              className="shirts"
              data-name="shirts"
              onClick={[Function]}
            >
              <h6
                data-name="shirts"
              >
                Shirts
              </h6>
            </div>
            <div
              className="tops"
              data-name="tops"
              onClick={[Function]}
            >
              <h6
                data-name="tops"
              >
                Tops
              </h6>
            </div>
            <div
              className="dresses"
              data-name="dresses"
              onClick={[Function]}
            >
              <h6
                data-name="dresses"
              >
                Dresses
              </h6>
            </div>
          </div>
        </div>
        <div
          className="filterCategory"
        >
          <h4
            className="categoryTitles"
          >
            Colour
          </h4>
          <div
            className="categoryTags"
          >
            <div
              data-name="black"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#101010",
                  "borderBottom": "2px solid #202020",
                  "color": "#C3C3C3",
                }
              }
            >
              <h6
                data-name="black"
              >
                Black
              </h6>
            </div>
            <div
              data-name="darkestSpruce"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#0E6251",
                  "borderBottom": "2px solid #0C5647",
                  "color": "#A5C4BE",
                }
              }
            >
              <h6
                data-name="darkestSpruce"
              >
                Darkest Spruce
              </h6>
            </div>
            <div
              className="multi"
              data-name="multi"
              onClick={[Function]}
            >
              <h6
                data-name="multi"
              >
                Multi
              </h6>
            </div>
            <div
              data-name="blue"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#3498DB",
                  "borderBottom": "2px solid #2E85C0",
                  "color": "#CCE5F6",
                }
              }
            >
              <h6
                data-name="blue"
              >
                Blue
              </h6>
            </div>
            <div
              data-name="blueDenim"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#546E7A",
                  "borderBottom": "2px solid #4A606B",
                  "color": "#D4DBDE",
                }
              }
            >
              <h6
                data-name="blueDenim"
              >
                Blue Denim
              </h6>
            </div>
            <div
              data-name="navy"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#2C3E50",
                  "borderBottom": "2px solid #212F3C",
                  "color": "#CACFD3",
                }
              }
            >
              <h6
                data-name="navy"
              >
                Navy
              </h6>
            </div>
            <div
              data-name="green"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#229954",
                  "borderBottom": "2px solid #1A733F",
                  "color": "#C8E6D4",
                }
              }
            >
              <h6
                data-name="green"
              >
                Green
              </h6>
            </div>
            <div
              data-name="white"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderBottom": "2px solid #DFDFDF",
                  "color": "#303030",
                }
              }
            >
              <h6
                data-name="white"
              >
                White
              </h6>
            </div>
            <div
              data-name="grape"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#884EA0",
                  "borderBottom": "2px solid #663B78",
                  "color": "#E1D3E7",
                }
              }
            >
              <h6
                data-name="grape"
              >
                Grape
              </h6>
            </div>
            <div
              data-name="khaki"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#AED581",
                  "borderBottom": "2px solid #83A061",
                  "color": "#6D8551",
                }
              }
            >
              <h6
                data-name="khaki"
              >
                Khaki
              </h6>
            </div>
            <div
              data-name="grey"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#BFC9CA",
                  "borderBottom": "2px solid #8F9798",
                  "color": "#777E7E",
                }
              }
            >
              <h6
                data-name="grey"
              >
                Grey
              </h6>
            </div>
            <div
              data-name="vintageIndigo"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#7986CB",
                  "borderBottom": "2px solid #5B6598",
                  "color": "#DDE1F2",
                }
              }
            >
              <h6
                data-name="vintageIndigo"
              >
                Vintage Indigo
              </h6>
            </div>
            <div
              data-name="mineralBlue"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#80DEEA",
                  "borderBottom": "2px solid #60A6AF",
                  "color": "#406F75",
                }
              }
            >
              <h6
                data-name="mineralBlue"
              >
                Mineral Blue
              </h6>
            </div>
            <div
              data-name="charcoal"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#424242",
                  "borderBottom": "2px solid #292929",
                  "color": "#D0D0D0",
                }
              }
            >
              <h6
                data-name="charcoal"
              >
                Charcoal
              </h6>
            </div>
            <div
              data-name="tan"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#CA6F1E",
                  "borderBottom": "2px solid #985316",
                  "color": "#F2DBC7",
                }
              }
            >
              <h6
                data-name="tan"
              >
                Tan
              </h6>
            </div>
            <div
              data-name="dustyBlue"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#AED6F1",
                  "borderBottom": "2px solid #83A1B5",
                  "color": "#6D8697",
                }
              }
            >
              <h6
                data-name="dustyBlue"
              >
                Dusty Blue
              </h6>
            </div>
            <div
              data-name="midwash"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#7FB3D5",
                  "borderBottom": "2px solid #5F86A0",
                  "color": "#DFECF5",
                }
              }
            >
              <h6
                data-name="midwash"
              >
                Midwash
              </h6>
            </div>
            <div
              data-name="red"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#E53935",
                  "borderBottom": "2px solid #AC2B28",
                  "color": "#F8CECC",
                }
              }
            >
              <h6
                data-name="red"
              >
                Red
              </h6>
            </div>
            <div
              data-name="rose"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#EB9696",
                  "borderBottom": "2px solid #B07171",
                  "color": "#FAE5E5",
                }
              }
            >
              <h6
                data-name="rose"
              >
                Rose
              </h6>
            </div>
            <div
              data-name="lilac"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#E1BEE7",
                  "borderBottom": "2px solid #A98FAD",
                  "color": "#F7EFF9",
                }
              }
            >
              <h6
                data-name="lilac"
              >
                Lilac
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="categoryTagsContainer"
      >
        <div
          className="female"
          data-name="female"
          onClick={[Function]}
        >
          <h6
            data-name="female"
          >
            WomensÂ Â Â Ã
          </h6>
        </div>
        <div
          className="jacket"
          data-name="jacket"
          onClick={[Function]}
        >
          <h6
            data-name="jacket"
          >
            Coats & JacketsÂ Â Â Ã
          </h6>
        </div>
        <div
          className="shirts"
          data-name="shirts"
          onClick={[Function]}
        >
          <h6
            data-name="shirts"
          >
            ShirtsÂ Â Â Ã
          </h6>
        </div>
        <div
          className="dresses"
          data-name="dresses"
          onClick={[Function]}
        >
          <h6
            data-name="dresses"
          >
            DressesÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="darkestSpruce"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#0E6251",
              "borderBottom": "2px solid #0C5647",
              "color": "#A5C4BE",
            }
          }
        >
          <h6
            data-name="darkestSpruce"
          >
            Darkest SpruceÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="blue"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#3498DB",
              "borderBottom": "2px solid #2E85C0",
              "color": "#CCE5F6",
            }
          }
        >
          <h6
            data-name="blue"
          >
            BlueÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="navy"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#2C3E50",
              "borderBottom": "2px solid #212F3C",
              "color": "#CACFD3",
            }
          }
        >
          <h6
            data-name="navy"
          >
            NavyÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="white"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderBottom": "2px solid #DFDFDF",
              "color": "#303030",
            }
          }
        >
          <h6
            data-name="white"
          >
            WhiteÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="khaki"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#AED581",
              "borderBottom": "2px solid #83A061",
              "color": "#6D8551",
            }
          }
        >
          <h6
            data-name="khaki"
          >
            KhakiÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="vintageIndigo"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#7986CB",
              "borderBottom": "2px solid #5B6598",
              "color": "#DDE1F2",
            }
          }
        >
          <h6
            data-name="vintageIndigo"
          >
            Vintage IndigoÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="charcoal"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#424242",
              "borderBottom": "2px solid #292929",
              "color": "#D0D0D0",
            }
          }
        >
          <h6
            data-name="charcoal"
          >
            CharcoalÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="dustyBlue"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#AED6F1",
              "borderBottom": "2px solid #83A1B5",
              "color": "#6D8697",
            }
          }
        >
          <h6
            data-name="dustyBlue"
          >
            Dusty BlueÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="red"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E53935",
              "borderBottom": "2px solid #AC2B28",
              "color": "#F8CECC",
            }
          }
        >
          <h6
            data-name="red"
          >
            RedÂ Â Â Ã
          </h6>
        </div>
        <div
          data-name="lilac"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E1BEE7",
              "borderBottom": "2px solid #A98FAD",
              "color": "#F7EFF9",
            }
          }
        >
          <h6
            data-name="lilac"
          >
            LilacÂ Â Â Ã
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
